aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,272 @@
+
+.. TODO in RC2 Middleware later in the request pipeline can access route data using extensions methods on ``HttpContext``. 
+
+To use routing, add it to the **dependencies** in ``project.json``:
+
+.. literalinclude:: routing/sample/RoutingSample/project.json
+  :linenos:
+  :dedent: 2
+  :language: javascript
+  :lines: 6-12
+  :emphasize-lines: 4
+  
+Add routing to ``ConfigureServices`` in ``Startup.cs``:
+
+.. literalinclude:: routing/sample/RoutingSample/Startup.cs
+  :linenos:
+  :dedent: 8
+  :language: c#
+  :lines: 16-19
+  :emphasize-lines: 3
+
+
+Configuring Routing
+-------------------
+Routing is enabled in the ``Configure`` method in the ``Startup`` class. Create an instance of `RouteBuilder <https://github.com/aspnet/Routing/blob/1.0.0-rc1/src/Microsoft.AspNet.Routing/RouteBuilder.cs>`_. You can optionally set the ``ServiceProvider`` and/or ``DefaultHandler`` properties, in order to make them available as you build routes.
+
+.. literalinclude:: routing/sample/RoutingSample/Startup.cs
+  :linenos:
+  :dedent: 8
+  :language: c#
+  :lines: 22-35
+  :emphasize-lines: 7-8,10-13
+  
+You can see on the last line above how ``ApplicationServices`` is used to access the ``IInlineConstraingResolver`` dependency; if this were done from a ``RouteBuilder`` extension method, access to the services would be done through the ``ServiceProvider`` property.
+
+Once you've finished adding routes to the ``RouteBuilder`` instance, call ``UseRouter`` and pass it the result of the ``RouteBuilder.Build`` method (demonstrated below).
+
+.. tip:: If you are only configuring a single route, you can simply call ``app.UseRouter`` and pass in the ``IRouter`` instance you wish to use, bypassing the need to use a ``RouteBuilder``.
+
+The route configured abovve will only match requests of the form "hello/{name}" where `name` is constrained to be alphabetical. Requests that match this will be handled by a custom ``IRouter`` implementation, ``HelloRouter``.
+
+.. literalinclude:: routing/sample/RoutingSample/HelloRouter.cs
+  :linenos:
+  :language: c#
+
+``HelloRouter`` checks to see if ``RouteData`` includes a value for the key ``name``. If not, it immediately returns without handling the request. Otherwise, the request is handled (by writing out "Hi {name}!") and the ``RouteContext`` is updated to note that the request was handled. This prevents additional routes from handling the request.
+
+This route was configured to use an inline constraint (see below), signified by the ":alpha" in the route template for the name route value. This constraint limits which requests this route will handle, in this case to alphabetical values for ``name``. Thus, a request for "/hello/steve" will be handled, but a request to "/hello/123" will not (instead, the request will pass through to the "Hello World!" request delegate).
+
```

Comment:
This is a pretty solid intro. If the idea here is to describe the execution flow of routes, it might be good to add a second route without the constraint and show that it will run for `/hello/123` but not `/hello/steve`